services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/models/shared_store.py

```diff
@@ -14,6 +23,44 @@ class SharedStore(BaseModel):
         description="stores the container names from the compose_spec",
     )
 
-    def clear(self):
+    # NOTE: setting up getter and setter does not work.
+    def set_shared_store_dir(self, shared_store_dir: Path) -> None:
+        self._shared_store_dir = shared_store_dir
+
+    async def clear(self):
         self.compose_spec = None
         self.container_names = []
+        await self.persist_to_disk()
+
+    @classmethod
+    async def init_from_disk(cls, shared_store_dir: Path) -> "SharedStore":
+        data_file_path = shared_store_dir / STORE_FILE_NAME
+        if data_file_path.exists():
+            # if the sidecar is started for a second time (usually the container dies)
+            # it will load the previous data which was stored
+            async with aiofiles.open(shared_store_dir / STORE_FILE_NAME) as data_file:
```

Comment:
isn't one of those pickle thing just made for dumping/loading from the disk? not sure never used it... but it looks a bit like it

Reply:
Not sure to what you are referring to, but pickle or json are equivalent in this situation. Will stick with this json encoding of the pydantic model I'm using. Going via pickle is much harder.